app/grandchallenge/algorithms/models.py

```diff
@@ -69,6 +69,38 @@
 JINJA_ENGINE = sandbox.ImmutableSandboxedEnvironment()
 
 
+class AlgorithmInterfaceManager(models.Manager):
+    def get_existing_interface_for_inputs_and_outputs(
+        self, *, inputs, outputs
+    ):
+        try:
+            return self.annotate(
```

Comment:
There might be a bug here that I think we faced in algorithm input matching before. I think the current query here would match when searching for `inputs=[A, B]` if the inputs on the object were `[A, C]`.

Reply:
You are completely right. I will update the annotation and filtering accordingly.